File: inventory_app/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String? sku;
  final String? description;
  final double price;
  final double cost;
  final int stock;
  final int minStock;
  final String unit;
  final String? supplier;

  Product({
    required this.id,
    required this.name,
    this.sku,
    this.description,
    required this.price,
    required this.cost,
    required this.stock,
    required this.minStock,
    required this.unit,
    this.supplier,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      sku: json['sku'],
      description: json['description'],
      price: (json['price'] ?? 0).toDouble(),
      cost: (json['cost'] ?? 0).toDouble(),
      stock: json['stock'] ?? 0,
      minStock: json['min_stock'] ?? 5,
      unit: json['unit'] ?? 'unidad',
      supplier: json['supplier'],
    );
  }
}

File: inventory-api/app/models/product.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.database import Base

class Product(Base):
    __tablename__ = "products"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    sku = Column(String, unique=True, nullable=True)
    description = Column(String, nullable=True)
    category_id = Column(Integer, ForeignKey("categories.id"), nullable=True)
    price = Column(Float, default=0.0)
    cost = Column(Float, default=0.0)
    stock = Column(Integer, default=0)
    min_stock = Column(Integer, default=5)
    unit = Column(String, default="unidad")  # unidad, kg, litro, caja, paquete, metro
    supplier = Column(String, nullable=True)
    image_url = Column(String, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    category = relationship("Category", backref="products")